source/s3_meta_request.c

```diff
@@ -1677,12 +1692,34 @@ void aws_s3_meta_request_cancel_ongoing_http_requests_synced(struct aws_s3_meta_
             AWS_CONTAINER_OF(request_node, struct aws_s3_request, ongoing_http_requests_list_node);
         if (!request->always_send) {
             /* Cancel the ongoing http stream, unless it's always send. */
-            aws_http_stream_cancel(request->synced_data.http_stream, error_code);
+            aws_http_stream_cancel(request->synced_data.ongoing_http_stream, error_code);
         }
-        request->synced_data.http_stream = NULL;
+        request->synced_data.ongoing_http_stream = NULL;
```

Comment:
debatable: seems accident-prone that we put `always_send` requests in this list, but then we need to be careful about not canceling.

Maybe rename ~ongoing_http_stream~ and ~ongoing_http_requests_list~ to `cancelable_http_stream` and `cancelable_http_requests_list`, and `always_send` requests shouldn't touch those variables at all